backend/Tim.Backend/Providers/Database/CouchbaseDbClient.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        /// <summary>
+        /// Connect to a couchbase collection.
+        /// </summary>
+        /// <typeparam name="T">Entity type.</typeparam>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public async Task<ICouchbaseCollection> CollectionAsync<T>()
+            where T : IJsonEntity
+        {
+            if (Bucket is null)
+            {
+                throw new InvalidOperationException("Bucket must be initialized.");
+            }
+
+            var collectionName = GetCollectionName<T>();
+            return await Bucket.CollectionAsync(collectionName);
+        }
+
+        /// <summary>
+        /// Get the default scope name for this bucket.
+        /// </summary>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public async Task<string> GetScopeName()
+        {
+            var scope = await Bucket.DefaultScopeAsync();
+            return scope.Name;
+        }
+
+        private async Task CreateCollectionIfNotExists<T>()
+        {
+            if (Bucket is null)
+            {
+                throw new InvalidOperationException("Bucket must be initialized.");
+            }
+
+            var collectionName = GetCollectionName<T>();
+            var scopeName = await GetScopeName();
+            var collectionSpec = new CollectionSpec(scopeName, collectionName);
+
+            try
+            {
+                var retries = 0;
+                while (retries++ < 3)
+                {
+                    await Bucket.Collections.CreateCollectionAsync(collectionSpec);
+                    await Task.Delay(TimeSpan.FromMilliseconds(retries * 500));
+                }
+            }
+            catch (CollectionExistsException)
+            {
+                Logger.Information($"Collection {collectionName} already exists.");
+            }
+
+            await CouchBaseClient.QueryIndexes.CreatePrimaryIndexAsync(
+                Bucket.Name,
+                new CreatePrimaryQueryIndexOptions()
+                    .ScopeName(scopeName)
+                    .CollectionName(collectionName)
+                    .IgnoreIfExists(true));
+        }
+
+        private async Task<IBucket> CreateBucketIfNotExists(string bucketName)
+        {
+            if (CouchBaseClient is null)
+            {
+                throw new InvalidOperationException("CouchBaseClient must be initialized with a connection.");
+            }
+
+            var buckets = await CouchBaseClient.Buckets.GetAllBucketsAsync();
+            if (!buckets.ContainsKey(bucketName))
+            {
+                await CouchBaseClient.Buckets.CreateBucketAsync(new BucketSettings
+                {
+                    BucketType = BucketType.Couchbase,
+                    Name = bucketName,
+                    NumReplicas = 0,
```

Comment:
should be const at top of class

Reply:
yeah, this is super awkward and should be handled at the couchbase level or at least kubernetes. I just had some issues initializing the buckets, so this is going to be temporary code until I can figure that out :(
will add a TODO note